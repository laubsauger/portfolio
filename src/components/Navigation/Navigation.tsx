import React from "react";
import DarkModeToggle from "../DarkModeToggle/DarkModeToggle";
import {Link} from "react-router-dom";

export default function Navigation() {
  return (
    <nav className="bg-gray-200 dark:bg-gray-800 grid grid-cols-2 w-full place-content-between items-center p-2">
      <Link to="/"><h4 className="text-gray-800 dark:text-gray-300">Portfolio</h4></Link>
      <DarkModeToggle />
    </nav>
  )
}