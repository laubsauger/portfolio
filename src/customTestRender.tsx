import React, {ReactElement} from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import { ThemeProvider } from "./contexts/ThemeContext";

type Props = {
  children?: ReactElement
}

const Wrapper: React.FC<Props> = ({children}) => {
  return (
    <ThemeProvider>
      <MemoryRouter>
        { children }
      </MemoryRouter>
    </ThemeProvider>
  );
};

const customRender = (ui:ReactElement, options?:any) => {
  return render(ui, { wrapper: Wrapper, ...options })
};

export * from "@testing-library/react";

export { customRender as render };