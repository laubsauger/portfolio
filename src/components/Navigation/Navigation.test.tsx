import * as React from 'react';
import {render} from "../../customTestRender";
import matchMedia from "../../matchMedia.mock";
import Navigation from "./Navigation";

describe('Navigation', () => {
  afterEach(() => {
    matchMedia.clear();
  });

  it('renders', () => {
    const { asFragment } = render(<Navigation />);
    expect(asFragment()).toMatchSnapshot();
  });
});
