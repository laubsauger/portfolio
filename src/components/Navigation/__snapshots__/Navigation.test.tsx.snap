// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders 1`] = `
<DocumentFragment>
  <nav
    class="bg-gray-200 dark:bg-gray-800 grid grid-cols-2 w-full place-content-between items-center p-2"
  >
    <a
      href="/"
    >
      <h4
        class="text-gray-800 dark:text-gray-300"
      >
        Portfolio
      </h4>
    </a>
    <div
      class="flex items-center justify-end cursor-pointer"
    >
      <div
        class="text-gray-800 dark:text-gray-300 mr-2"
      >
        dark
      </div>
      <div>
        <svg
          color="white"
          fill="none"
          height="36"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="cursor: pointer; transform: rotate(40deg);"
          viewBox="0 0 24 24"
          width="36"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            id="circle-mask-1"
          >
            <rect
              fill="white"
              height="100%"
              width="100%"
              x="0"
              y="0"
            />
            <circle
              fill="black"
              r="9"
            />
          </mask>
          <circle
            cx="12"
            cy="12"
            fill="white"
            mask="url(#circle-mask-1)"
          />
          <g
            stroke="currentColor"
            style="opacity: 0;"
          >
            <line
              x1="12"
              x2="12"
              y1="1"
              y2="3"
            />
            <line
              x1="12"
              x2="12"
              y1="21"
              y2="23"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="4.22"
              y2="5.64"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="18.36"
              y2="19.78"
            />
            <line
              x1="1"
              x2="3"
              y1="12"
              y2="12"
            />
            <line
              x1="21"
              x2="23"
              y1="12"
              y2="12"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="19.78"
              y2="18.36"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="5.64"
              y2="4.22"
            />
          </g>
        </svg>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
